3rd party/StashPlexAgent.bundle/Contents/Code/__init__.py

```diff
@@ -83,110 +71,241 @@ def search(self, results, media, lang):
         data = HttpReq(id_query % mid)
         data = data['data']['findScene']
         metadata.collections.clear()
-      
-        if data['date']:
-            try:
-                Log("Trying to parse:" + data["date"])
-                date=dateparser().parse(data["date"])
-            except Exception as ex:
-                Log(ex)
-                date=None
-                pass
-            # Set the date and year if found.
-            if date is not None:
-                metadata.originally_available_at = date
-                metadata.year = date.year
-
-        # Get the title
-        if data['title']:
-            metadata.title = data["title"]
-
-        # Get the Studio
-        if not data["studio"] is None:
-            metadata.studio = data["studio"]["name"]
-        
-        # Get the rating
-        if not data["rating"] is None:
-            metadata.rating = float(data["rating"]) * 2
-            if Prefs["CreateRatingTags"]:
-                if int(data["rating"]) > 0:
-                    rating = str(int(data["rating"]))
-                    ratingstring = "Rating: " + rating + " Stars"
-                    try:
-                        metadata.collections.add(ratingstring)
-                    except:
-                        pass                  
-        
-        # Set the summary
-        if data['details']:
-            summary=data["details"].replace("\n"," ").replace("\r", "").replace("\t","")
-            metadata.summary = summary
-
-        # Set series and add to collections
-        if Prefs["CreateCollectionTags"]:
-            if not data["studio"] is None:
-                site="Site: " + data["studio"]["name"]
-                try:
-                    metadata.collections.add(site)
-                except:
-                    pass
-                if not data["studio"]["parent_studio"] is None:
-                    site="Studio: " + data["studio"]["parent_studio"]["name"]
+
+        allow_scrape = False
+        if (Prefs["RequireOrganized"] and data["organized"]) or not Prefs["RequireOrganized"]:
+            if DEBUG and Prefs["RequireOrganized"]:
+                Log("Passed 'Organized' Check, continuing...")
+            if (Prefs["RequireURL"] and data["url"]) or not Prefs["RequireURL"]:
+                if DEBUG and Prefs["RequireURL"]:
+                    Log("Passed 'RequireURL' Check, continuing...")
+                if (Prefs["RequireStashID"] and len(data["stash_ids"])) or not Prefs["RequireStashID"]:
+                    if DEBUG and Prefs["RequireStashID"]:
+                        Log("Passed 'RequireURL' Check, continuing...")
```

Comment:
typo ? `RequireURL` -> `RequireStashID`

Reply:
Yeah, in my prefs I have those two options...  guessing you can tell the order I copy/pasted.  :D